lib/pages/Notifications.dart

```diff
@@ -1,143 +1,191 @@
-            crossAxisAlignment: CrossAxisAlignment.start,
-            children: <Widget>[
-              Padding(
-                padding: EdgeInsets.only(left: 15.0, top: 5.0, bottom: 8.0),
-                child: Text('Notifications',
-                    style: CarriageTheme.largeTitle),
-              ),
-              Detail(),
-              Container(
-                color: Colors.white,
-                height: MediaQuery.of(context).size.height / 5,
-              )
-            ]),
-      )
+    Widget icon = CircleAvatar(
+        radius: 22,
+        child: CircleAvatar(
+          radius: 22,
+          backgroundColor: color,
+          foregroundColor: Colors.white,
+          child: Icon(iconData),
+        )
     );
+    return Notification(this.ride, this.notifTime, icon, 'Dispatcher', this.text);
   }
 }
 
-class Detail extends StatefulWidget {
+class Notification extends StatelessWidget {
+  Notification(this.ride, this.notifTime, this.circularWidget, this.title, this.text);
+
+  final Ride ride;
+  final DateTime notifTime;
+  final Widget circularWidget;
+  final String title;
+  final String text;
+
   @override
-  _DetailState createState() => _DetailState();
-}
+  Widget build(BuildContext context) {
+    DateTime now = DateTime.now();
+    Duration timeSinceNotif = now.difference(notifTime);
+    String time = '';
+
+    bool timeWithin(Duration threshold) {
+      return timeSinceNotif.compareTo(threshold) <= 0;
+    }
+
+    if (timeWithin(Duration(hours: 1))) {
+      time = '${timeSinceNotif.inMinutes} min ago';
+    }
 
-class _DetailState extends State<Detail> {
-
-  List<Color> colors = [
-    Colors.black,
-    Colors.black,
-    Colors.black,
-    Colors.red,
-    Colors.black,
-    Colors.red
-  ];
-
-  List<IconData> icons = [
-    Icons.directions_car,
-    Icons.check,
-    Icons.directions_car,
-    Icons.edit,
-    Icons.directions_car,
-    Icons.cancel
-  ];
-
-  List<String> text = [
-    'Your driver is on the way! Wait outside to meet your driver.',
-    'Your ride on MM/DD/YYYY has been confirmed',
-    'How was your ride with Jennifer? Leave a review so we can improve our service',
-    'Your ride information has been edited by the admin. Please review your ride info.',
-    'Your driver is here! Meet your driver at the pickup point.',
-    'Your driver cancelled the ride because the driver was unable to find you.'
-  ];
-
-  Widget notificationRow(BuildContext context, Color color, IconData icon, String text) {
-    return Padding(
-        padding: EdgeInsets.only(top: 10, bottom: 10),
+    else if (timeWithin(Duration(days: 1))) {
+      int hoursSince = timeSinceNotif.inHours;
+      time = '$hoursSince hour${hoursSince == 1 ? '' : 's'} ago';
+    }
+    else if (timeWithin(Duration(days: 3))) {
+      int daysSince = timeSinceNotif.inDays;
+      time = '$daysSince day${daysSince == 1 ? '' : 's'} ago';
+    }
+    else {
+      time = DateFormat('yMd').format(notifTime);
+    }
```

Comment:
this should be okay imo, but maybe a library would be better to handle this?

Reply:
I'm not sure if there are any libraries that fit it exactly--https://pub.dev/packages/time might make calculations easier but doesn't really help with the whole if/else statement if that's what you mean?

Reply:
yeah i was mostly referring to the if/else. if there doesnt seem to be any good libraries then it should be fine